deploy/helm-chart/README.md

```diff
@@ -0,0 +1,81 @@
+
+1. Clone the repo and change into the `nginx-kubernetes-gateway` directory:
+
+   ```
+   git clone https://github.com/nginxinc/nginx-kubernetes-gateway.git
+   cd nginx-kubernetes-gateway
+   ```
+
+1. Install the Gateway API resources from the standard channel (the CRDs and the validating webhook):
+
+   ```
+   kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v0.7.1/standard-install.yaml
+   ```
+
+1. Create the nginx-gateway Namespace:
+
+    ```
+    kubectl create ns nginx-gateway
+    ```
+
+### Installing the Chart from local
+
+To install the chart with the release name my-release (my-release is the name that you choose) into the nginx-gateway
+namespace:
+
+```
+helm install my-release ./deploy/helm-chart -n nginx-gateway
+```
+
+### Uninstalling the Chart
+
+To uninstall/delete the release `my-release`:
+
+```
+helm uninstall my-release -n nginx-gateway
+```
+
+The command removes all the Kubernetes components associated with the release and deletes the release.
+
+## Configuration
+
+The following tables lists the configurable parameters of the NGINX Kubernetes Gateway chart and their default values.
+
+|Parameter | Description | Default |
+| --- | --- | --- |
+|`gateway.nginxGateway.image.repository` | The repository for the NGINX Kubernetes Gateway image. | ghcr.io/nginxinc/nginx-kubernetes-gateway |
+|`gateway.nginxGateway.image.tag` | The tag for the NGINX Kubernetes Gateway image. | edge |
+|`gateway.nginxGateway.imagePullPolicy` | The imagePullPolicy for the NGINX Kubernetes Gateway image. | Always |
+|`gateway.nginxGateway.gatewayClass` | The GatewayClass for the NGINX Kubernetes Gateway deployment. | nginx |
+|`gateway.nginx.image.repository` | The repository for the NGINX image. | nginx |
+|`gateway.nginx.image.tag` | The tag for the NGINX image. | 1.25 |
+|`gateway.nginx.imagePullPolicy` | The imagePullPolicy for the NGINX image. | Always |
+|`gateway.initContainer.image.repository` | The repository for the initContainer image. | busybox |
+|`gateway.initContainer.image.tag` | The tag for the initContainer image. | 1.36 |
+|`gateway.serviceAccount.annotations` | Annotations for the ServiceAccount used by the NGINX Kubernetes Gateway deployment. | {} |
+|`gateway.serviceAccount.name` | Name of the ServiceAccount used by the NGINX Kubernetes Gateway deployment. | Autogenerated |
+|`gateway.service.create` | Creates a service to expose the NGINX Kubernetes Gateway pods. | true |
+|`gateway.service.type` | The type of service to create for the NGINX Kubernetes Gateway. | Loadbalancer |
+|`gateway.service.externalTrafficPolicy` | The externalTrafficPolicy of the service. The value Local preserves the client source IP. | Local |
+|`gateway.service.annotations` | The annotations of the NGINX Kubernetes Gateway service. | true |
+|`gateway.service.httpPort.enable` | Enables the HTTP port for the NGINX Kubernetes Gateway service. | true |
```

Comment:
Could  we just have `ports` (renamed `customPorts`) configured by default with http and https ports? This way we will have less parameters. Also, as opposed to NIC, NKG doesn't really have the default ports - ports are configured through the Gateway resources. At the same time, in our examples, we use `http 80` and `https 443` ports in the example Gateways, that's why we included them in the Service manifests.

Reply:
I see your point, and I have changed it, but it does go against the best practices (see https://helm.sh/docs/chart_best_practices/values/#consider-how-users-will-use-your-values). I guess that's probably OK in this case though.